app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -94,6 +97,14 @@ abstract class BaseScreen<K : ScreenKey, B : ViewBinding, M : Parcelable, E, F,
     })
 
     _viewModel.models.observe(viewLifecycleOwner, uiRenderer()::render)
+
+    _viewModel.viewEffects.setObserver(
+        viewLifecycleOwner,
+        viewEffectHandler()::handle,
+        { viewEffects ->
+          viewEffects.forEach(viewEffectHandler()::handle)
+        }
```

Comment:
Can we extract this to a named function (like `runQueuedViewEffects` or something) so that the intention is more clear?

Reply:
If we can use Kotlin named parameters, that should work as well.

Reply:
This is a Java method, so we cannot use named params. Will extract this into a named function.

Reply:
Updated it. Moved the handling into separate methods, we have to handle 2 types of view effects. One is live view effect, when the app is in foreground. When app is in background Mobius will add the view effects in to queue and trigger them once app is in foreground again. That will happen in paused view effects observer.
![image](https://user-images.githubusercontent.com/6140516/126592337-35f89e65-48c3-4b5a-98d6-13e10afced0e.png)

Reply:
Hmm, can't we just delegate to `runLiveViewEffect` for the paused effects instead of creating a new method?

```kotlin
pausedViewEffects.forEach { runLiveViewEffect(viewEffectHandler, it) }
```

Reply:
Nitpick: `runLiveViewEffects` should be singular since we will only execute one effect.

Reply:
I didn't want to use `runLiveViewEffect` inside the pause view effects loop, so created a new method. We can just go back to using the `viewEffectHandler` method references directly and have named params for the observer result like we do now?

Reply:
This
```kt
val viewEffectHandler = viewEffectHandler()
_viewModel.viewEffects.setObserver(
    viewLifecycleOwner,
    { liveViewEffect -> viewEffectHandler.handle(liveViewEffect) },
    { pausedViewEffects -> pausedViewEffects.forEach(viewEffectHandler::handle) }
)
```

Reply:
I like this approach.